src/test/java/net/datafaker/ArchitectureTest.java

```diff
@@ -0,0 +1,45 @@
+package net.datafaker;
+
+import com.tngtech.archunit.core.domain.JavaClasses;
+import com.tngtech.archunit.core.importer.ClassFileImporter;
+import com.tngtech.archunit.lang.ArchRule;
+import net.datafaker.annotations.InternalApi;
+import net.datafaker.idnumbers.IdNumberGenerator;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import static com.tngtech.archunit.base.DescribedPredicate.not;
+import static com.tngtech.archunit.core.domain.JavaClass.Predicates.type;
+import static com.tngtech.archunit.core.importer.ImportOption.Predefined.DO_NOT_INCLUDE_TESTS;
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.classes;
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.methods;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+
+@TestInstance(PER_CLASS)
+public class ArchitectureTest {
+    /*
+     * 1. Don't hold these in static fields - otherwise they will consume memory till the end of all tests
+     * 2. Use `@TestInstance(PER_CLASS)` to reuse they between tests
+     */
+    private final JavaClasses PRODUCTION_CLASSES = new ClassFileImporter().withImportOption(DO_NOT_INCLUDE_TESTS).importPackages("net.datafaker");
+
+    @Test
+    public void packagePrivateMethodsShouldBeMarkedAsInternalAPI() {
+        ArchRule rule = methods()
+            .that().arePackagePrivate()
+            .should().beAnnotatedWith(InternalApi.class);
```

Comment:
It was not a question to me, but still I feel urge to share that it might be an overkill. Apologies 😎 

1) now we do this to enforce users to treat `package-private` as a pattern to narrow down a unit to test
2) then (I'm speculating now) someone will want a true `package-private` feature and he'll need to introduce a new annotation `@PackagePrivateApi`
3) a new contributor comes and gets scared away

So, I'd think this steeps up learning curve for a newcomer. 
Even if implemented the code still could be misused,
- previously one would misuse API and package naming
- after this PR one would misuse annotations